lib/modules/datasource/docker/index.ts

```diff
@@ -844,18 +861,25 @@ export class DockerDatasource extends Datasource {
         });
       } catch (err) {
         if (
-          !foundMaxResultsError &&
+          !foundMaxResults &&
           err instanceof HttpError &&
           isECRMaxResultsError(err)
         ) {
           const maxResults = 1000;
           url = `${registryHost}/${dockerRepository}/tags/list?n=${maxResults}`;
           url = ensurePathPrefix(url, '/v2');
-          foundMaxResultsError = true;
+          foundMaxResults = true;
           continue;
         }
         throw err;
       }
+      if (!foundMaxResults && isECRMaxResultsResponse(res)) {
```

Comment:
This is totally wrong, as res is undefined or the result of the prevous request. So revert